routes/admin.js

```diff
@@ -1,9 +1,24 @@
 const express = require("express")
-
+const multer = require("multer")
 const adminController = require("../controllers/adminController")
 
+const storageConfig = multer.diskStorage({
+  destination: function(req, file, cb) {
+    cb(null, "public/images")
+  },
+  filename: function(req, file, cb) {
+    cb(null, file.originalname)
+  }
+})
+
+const upload = multer({storage: storageConfig})
+
 const router = express.Router()
 
 router.get("/admin", adminController.viewAdmin)
 
+router.get("/add", adminController.addProductView)
+
+router.post("/add", upload.single("image"), adminController.addProduct)
```

Comment:
What happens to the image if the validation of the product fails? Is it still saved to the server?

Is there a way to validate the input is correct before saving the file?